src/appkit.cs

```diff
@@ -19275,8 +19275,17 @@ partial interface NSTextView : NSTextInputClient, NSTextLayoutOrientationProvide
 		[Export ("toggleSmartInsertDelete:")]
 		void ToggleSmartInsertDelete (NSObject sender);
 
+#if !XAMCORE_4_0
+		[Obsolete ("Use 'SmartInsert(string, NSRange, out string, out string)' overload instead.")]
 		[Export ("smartInsertForString:replacingRange:beforeString:afterString:")]
 		void SmartInsert (string pasteString, NSRange charRangeToReplace, string beforeString, string afterString);
+#endif
+
+#if !XAMCORE_4_0
+		[Sealed] // sealed until the obsolete overload has been removed
```

Comment:
Does the deprecated `SmartInsert` work ? or does it **always** crash (like mentioned in https://github.com/xamarin/xamarin-macios/issues/8349) ?

If it **never** work you can remove the `[Sealed]` so the new, working API is not limited (i.e. can be overridden) even before `XAMCORE_4_0`.

For that to build without error you must remove the `[Export]` on old, non-working, binding. You can use `[Wrap ("throw new NotSupportedException ()")]` or similar for its implementation.


Also the PR message links to the PR itself, not to the https://github.com/xamarin/xamarin-macios/issues/8349 issue.

Reply:
Fixed PR link.

I believe it crashes unconditionally, but I need to try it locally.

Reply:
My [test](https://gist.github.com/chamons/1c029a21931079036b0f8c1c9db54afa) shows unconditionally crashing.

I'll update the binding as suggested, that's even better than expected.